Added tests and fixed the eval thing (Issue #1)

blottn opened the pull request:
Uhhh ignore how many commits this took, it's fine. 

Functionality is now simplified

```
var loc = require('localscope');
exec(loc({a: 12}));
console.log(a)    // 12
```

also optimised, only one eval now!

414owen commented:
Hmmm, I'm afraid you'd have to convince me that stringifying every property is faster than including the reference into the eval.

What I thought you were going to do was to still pass the name of the object to localscope, and to generate the string that declares the variables using the property references. That way you'd avoid the nested eval and the stringify. However I see now that this isn't possible, as iterating over the properties of the function requires the reference, rather than the name.

Also, there are things in Javascript that can't be converted to JSON, which would fail with this method. Functions, for instance.

blottn commented:
It couldn't declare functions previously though :P? also I think stringify is way cheaper than eval

blottn commented:
Added function support, removed stringify and reduced to only one eval. However there is now 2 parameters the object itself and the identifier of the object.
Example of new operation:
```
> const loc = require('localscope');
> var declare_these  = {
    a : 123,
    b : 'hello world,
    c : {
         inner : 'thing'
    },
    d : function() {
        return 'A function was called!';
    },
}

> eval(loc(declare_these,'declare_these'));
> a    // 123
> b    //  'hello world'
> c    //  { inner : 'thing' }
> d() //  'A function was called!'

```
